cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py

```diff
@@ -96,3 +109,81 @@ def test_provision_cloud_init(appliance, request, setup_provider, provider, prov
     with ssh.SSHClient(hostname=connect_ip, username=provisioning['ci-username'],
                        password=provisioning['ci-pass']) as ssh_client:
         wait_for(ssh_client.uptime, num_sec=200, handle_exception=True)
+
+
+@pytest.mark.rhv3
+@pytest.mark.uncollectif(lambda provider, appliance:
+    not provider.one_of(RHEVMProvider) or appliance.version < '5.10.0.12',
+    reason='Required customization template is only available in CFME 5.10.0.12 or greater.')
+def test_provision_cloud_init_payload(appliance, request, setup_provider, provider, provisioning,
+                                      vm_name):
+    """
+    Tests that options specified in VM provisioning dialog in UI are properly passed as a cloud-init
+    payload to the newly provisioned VM.
+
+    Metadata:
+        test_flag: cloud_init, provision
+    """
+    image = provisioning.get('ci-image', None)
+    if not image:
+        pytest.skip('No ci-image found in provider specification.')
+    note = ('Testing provisioning from image {} to vm {} on provider {}'.format(
+        image, vm_name, provider.key))
+    logger.info(note)
+
+    ci_payload = {
+        'root_password': 'mysecret',
+        'address_mode': 'Static',
+        'hostname': 'cimachine',
+        'ip_address': '10.1.1.10',
```

Comment:
These need to be non-routable or defined within and pulled from yamls.

Please see the other tests here for how payload is defined in the yamls and resolved for tests.

Reply:
What about this?
ip_address: 169.254.0.1
subnet_mask: 29
gateway: 169.254.0.2
dns_servers: 169.254.0.3

This should not be routable as specified in this IANA doc: https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml

That request I understand. But why should we need to have this specified in yamls, given that this is just a dummny configuration? No matter in which environment you run it, it should work the same.